src/app/features/worklog/worklog-export/worklog-export.component.ts

```diff
@@ -164,54 +171,123 @@ export class WorklogExportComponent implements OnInit, OnDestroy {
-  private _createRows(tasks: WorklogTask[], startTimes: WorkStartEnd, endTimes: WorkStartEnd): RowItem[] {
-    const days: { [key: string]: RowItem } = {};
-    const _mapTaskToDay = (task, dateStr) => {
-      let day: RowItem = days[dateStr];
+  private _createRows(tasks: WorklogTask[], startTimes: WorkStartEnd, endTimes: WorkStartEnd, groupBy: WorklogGrouping): RowItem[] {
 
-      if (!day) {
-        day = days[dateStr] = {
-          date: dateStr,
+    const _mapToGroups = (task: WorklogTask) => {
+      const taskGroups: {[key: string]: RowItem} = {};
+      const createEmptyGroup = () => {
+        return {
+          dates: [],
           timeSpent: 0,
           timeEstimate: 0,
           tasks: [],
           titlesWithSub: [],
-          workStart: startTimes[dateStr],
-          workEnd: endTimes[dateStr],
+          titles: [],
+          workStart: undefined,
+          workEnd: undefined,
         };
-      }
-      days[dateStr] = {
-        ...days[dateStr],
-        timeSpent: day.timeSpent + task.timeSpentOnDay[dateStr],
-        timeEstimate: day.timeEstimate + task.timeEstimate,
-        tasks: [...day.tasks, task],
       };
-    };
 
-    tasks.forEach(task => {
-      if (task.subTaskIds && task.subTaskIds.length > 0) {
-      } else {
-        if (task.timeSpentOnDay) {
-          Object.keys(task.timeSpentOnDay).forEach(dateStr => {
-            _mapTaskToDay(task, dateStr);
+      // If we're grouping by parent task ignore subtasks
+      // If we're grouping by task ignore parent tasks
+      if ( (groupBy === WorklogGrouping.PARENT && task.parentId !== null)
+        || ( groupBy === WorklogGrouping.TASK && task.subTaskIds.length > 0 )
+      ) {
+          return taskGroups;
+      }
+
+      switch (groupBy) {
+        case WorklogGrouping.DATE:
+          if (!task.timeSpentOnDay) {
+            return {};
+          }
+          const numDays = Object.keys(task.timeSpentOnDay).length;
+          Object.keys(task.timeSpentOnDay).forEach(day => {
+            taskGroups[day] = {
+              dates: [day],
+              tasks: [task],
+              workStart: startTimes[day],
+              workEnd: endTimes[day],
+              timeSpent: 0,
+              timeEstimate: 0
+            };
+            if (!task.subTaskIds || task.subTaskIds.length === 0) {
+              taskGroups[day].timeSpent = task.timeSpentOnDay[day];
+              taskGroups[day].timeEstimate = task.timeEstimate / numDays;
+            }
+          });
+          break;
+        case WorklogGrouping.PARENT:
+        case WorklogGrouping.TASK:
+          taskGroups[task.id] = createEmptyGroup();
+          taskGroups[task.id].tasks = [task];
+          taskGroups[task.id].dates = Object.keys(task.timeSpentOnDay);
+          taskGroups[task.id].timeEstimate = task.timeEstimate;
+          taskGroups[task.id].timeSpent = Object.values(task.timeSpentOnDay).reduce((acc, curr) => acc + curr, 0);
```

Comment:
@johannesjo Previously I had been using task.timeSpent instead. However, as it turns out, `task.timeSpent` doesn't hold the _total_ time spent on a task, but only the time spent on one (the last?) day.

Is this intended behaviour?